Astro 302 - Taskflow Example Update

im-cr opened the pull request:
**TASK**
- Add associated animated gifs to each page via the new Lightbox implementation
- Use uploaded video with animated gifs as fallback in a <video> element

**IMPACTED AREAS**

**GRM Service UX Design**
- grm-service-ux-design/grm-dashboard/
- grm-service-ux-design/grm-equipment-manager/
- grm-service-ux-design/grm-schedule/

**TT&C Service UX Design**
- ttc-service-ux-design/about-the-ttc-designs/
- ttc-service-ux-design/ttc-monitor/
- ttc-service-ux-design/ttc-command/
- ttc-service-ux-design/ttc-investigate/

sknep commented on 404.md:
```diff
@@ -2,7 +2,6 @@
 layout: interior.template.njk
 title: Oops! Not Found
 permalink: 404.html
-eleventyExcludeFromCollections: true
```
can i ask why these lines got removed? did you branch from somewhere else? these are used to exclude pages from sitemap generation

sknep commented on _content/404.md:
```diff
@@ -2,7 +2,6 @@
 layout: interior.template.njk
 title: Oops! Not Found
 permalink: 404.html
-eleventyExcludeFromCollections: true
```
I think this needs to stay

sknep commented on _content/_data/meta.json:
```diff
@@ -1,3 +1,3 @@
 {
-	"version": "4.6"
```
this is going back in time -- possibly started from an old branch.

sknep commented on _content/_data/navigation.json:
```diff
@@ -264,25 +261,6 @@
-				"url": "/community/content-policy"
-			}
-		]
-	},
```
I think it's important that we lose none of these changes.

sknep commented on _content/community/content-policy.md:
```diff
@@ -1,87 +0,0 @@
----
```
This needs to come back

sknep commented on _content/grm-service-ux-design/grm-dashboard.md:
```diff
@@ -83,7 +83,14 @@ If operators choose to drill into an alert via the Investigate button in the Ale
 Below is an animated walkthrough of a representative task flow using the GRM Dashboard app. In this flow, the operator notices a critical alert for an upcoming contact, investigates it via the Contact Alert Details page and then resolves the issue by selecting an alternative piece of equipment to use for the contact.
 
-![Task Flow Example](/img/service-specific-ux-design/gif-placeholder.png)
+<div markdown="1">
```
It works, though a custom shortcode in 11ty might be cleaner.

sknep commented on _includes/components.template.njk:
```diff
@@ -32,7 +32,7 @@
 					</a>
 				{% endif %}
 				{% if git %}
-					<a class="sample-links__bitbucket" href="https://github.com/RocketCommunicationsInc/astro-componentssrc/components/{{git}}">Bitbucket</a>
```
Nope, we want github links, not bitbucket ones

sknep commented on css/legacy/chapters_legacy.css:
```diff
@@ -0,0 +1,181 @@
+main {
```
Are these files necessary?

sknep commented on netlify.toml:
```diff
@@ -68,10 +68,4 @@
 	to = "/ttc-service-ux-design/ttc-investigate"
 	status = 300
-	force = true
-
```
We still need this, we renamed the component

sknep commented on package.json:
```diff
@@ -9,7 +9,7 @@
 	"main": "index.js",
 	"scripts": {
 		"test": "echo \"Error: no test specified\" && exit 1",
-		"sass": "node-sass -rw scss -o css --output-style compressed",
```
AH, so this is why the css files are un-minified in this PR! Is there a reason? if not, they're adding a ton of noise to your PR

sknep commented on _includes/base.template.njk:
```diff
@@ -44,5 +45,7 @@
 			</div>
 		</div>
+
+		<script src="/js/lightbox.js"></script>
```
This loads the javascript for the lightbox on all pages -- is it used on _every_ page? If not, you could move it (and the stylesheet link) to just the template(s) where it shows up, keeping the site a little faster on non-lightboxed pages (compliance, home, etc)

dmcalester commented:
Hey Cortney, 

- I think most of @sknep’s comments are a result of this PR request to draft, it should be to master as these changes are fixes that should be up before draft is approved. But please confirm changing the target branch fixes all comments.
- The poster frame for the images should be a static poster image with a play button/icon (see https://draft--astrouxds.netlify.app/fd-service-ux-design/about-the-fd-designs/) for an example
- Sorry I did not make it clear, but the URLs for all MP4/WebM videos should reference the AWS links posted above

sknep commented:
This is entirely optional, but you might be interested in a CSS-only lightbox implementation. There are a few ways to do this, here's one: https://codepen.io/gschier/pen/HCoqh 

The implementations I've seen use in-page anchor links and `:target` to handle styling the image as, typically, position: absolute, with the scrim (semitransparent black background) on a pseudo-element of the container. When a colleague first showed me this, I had serious doubts, but it surprised me, and I wanted to pass along the neat idea in case you had an interest.

I agree with Duncan, there really needs to be some sort of play button or call to action over or inside the movie's poster image (which usually shows up if someone is on mobile, because videos on mobile don't usually autoplay). Not sure if that was provided to you, as the link Duncan sent above doesn't show a working video :(

im-cr commented:
@sknep good call on the css lightbox. Added that in with a small tweak around sizing of the lightbox content. Thanks. @dmcalester switched over to master from draft. Updated video paths. 

Also added in 'play' icon over existing placeholder gifs. Do we want to use a static png rather than animated gif placeholder? If not, I can add in small hover effect to better illustrate interaction. Let me know what you both think.

sknep commented on _content/grm-service-ux-design/grm-dashboard.md:
```diff
@@ -83,7 +83,21 @@ If operators choose to drill into an alert via the Investigate button in the Ale
+			alt="GRM Dashboard demo" />
+		</a>
+	</figure>
+	<a href="#" class="lightbox" id="demo" markdown="1">
```
When I close the lightbox (by clicking the scrim), it shoots me back up to the top of the page in chrome. Maybe call this `#close` so that it doesn't jump the page when you exit the lightbox?

dmcalester replied:
Static image placeholder

sknep commented on _content/grm-service-ux-design/grm-dashboard.md:
```diff
@@ -83,7 +83,21 @@ If operators choose to drill into an alert via the Investigate button in the Ale
+		</a>
+	</figure>
+	<a href="#" class="lightbox" id="demo" markdown="1">
+		<video width="99%" autoplay loop muted markdown="1" controls
```
I think the controls look a little weird on this, since its already autoplaying and looping. Did the designer really want the control bar?

sknep commented on scss/lightbox.scss:
```diff
@@ -0,0 +1,392 @@
+// .lb-loader,.lightbox{
```
That's a lot to comment out -- if we don't need it, feel free to remove

sknep commented on scss/lightbox.scss:
```diff
@@ -0,0 +1,392 @@
+//   opacity: 1;
+// }
+
+.lightbox{
```
I love it! so short and sweet

sknep commented:
> Also added in 'play' icon over existing placeholder gifs. Do we want to use a static png rather than animated gif placeholder? If not, I can add in small hover effect to better illustrate interaction. Let me know what you both think.

Something with a play button probably shouldn't be _moving_ underneath it. I think static png is better, personally. But was the original ask to implement both gif and mp4? I'm wondering why the designer gave you both assets. 

Hover effect sounds shiny! I'm all for seeing it :)

sknep commented:
Grr, the iFramed sales/support chat can sit on top of the video at some sizes. Probably not worth fixing but it is annoying:
![Screen Shot 2020-07-28 at 5 22 56 PM](https://user-images.githubusercontent.com/4451344/88728350-1f5e6180-d0f7-11ea-97eb-242add64c627.png)

sknep commented on _content/grm-service-ux-design/grm-dashboard.md:
```diff
@@ -83,7 +83,21 @@ If operators choose to drill into an alert via the Investigate button in the Ale
+	<a href="#" class="lightbox" id="demo" markdown="1">
+		<video width="99%" autoplay loop muted markdown="1" controls
+		poster="/img/service-specific-ux-design/grm-dashboard-modify-string.gif">
+			<source src="http://com.rocketcom.astrouxds.s3.us-west-2.amazonaws.com/media/grm-dashboard-modify-string.mp4" type="video/mp4" markdown="1">        
```
These all need to be at https, to avoid a security console error

sknep replied:
It looks like we have an SSL error here too:
`GET https://com.rocketcom.astrouxds.s3.us-west-2.amazonaws.com/media/grm-equipment-manager-sched-job.webm net::ERR_CERT_COMMON_NAME_INVALID`

dmcalester replied:
@im-cr @sknep getting the S3 bucket an SSL in the immediate future is going to be tough. There are some entanglements that make it a non-trivial lift. 

For now let’s revert to the animated gif walkthroughs and remove the videos. I’ll work on getting either the SSL situation sorted out or an alternate option.

dmcalester commented:
Minor rendering bug in Firefox 78.0.2 Play icon appears below the target image covering source materials
<img width="985" alt="Screen Shot 2020-08-07 at 12 54 03 PM" src="https://user-images.githubusercontent.com/60529018/89683358-3f3b1580-d8ad-11ea-8392-dd9dc25f97c7.png">

sknep commented:
There's something up with the branches on this repo -- 154 commits in this PR, some from Jeff in February? If this is ready to ship, I won't hold it up, but maybe this repo would benefit from a branch and merge strategy that keeps things cleaner.
